Resources/public/js/views/services/plugins/ez-dashboardblockallcontentserviceplugin.js

```diff
@@ -0,0 +1,310 @@
+         * @param result {Array} array of struct objects containing a `content` property
+         */
+        _setContentList: function (target, error, result) {
+            if (error) {
+                this._handleError({
+                    message: this.get('loadContentSearchResultsErrorMessage'),
+                    identifier: this.get('loadContentSearchResultsErrorIdentifier')
+                }, error);
+
+                return;
+            }
+
+            target.set('contentList', result);
+        },
+
+        /**
+         * Loads content root location
+         *
+         * @method _loadRootLocation
+         * @protected
+         * @return {Y.Promise}
+         */
+        _loadRootLocation: function () {
+            var errorData = {
+                message: this.get('loadRootLocationErrorMessage'),
+                identifier: this.get('loadRootLocationErrorIdentifier')
+            };
+
+            return new Y.Promise(Y.bind(function (resolve, reject) {
+                this.get('host').get('capi').getDiscoveryService().getInfoObject('rootLocation', function (error, response) {
+                    if (error) {
+                        reject(errorData);
+
+                        return;
+                    }
+
+                    try {
+                        resolve(response._href);
+                    } catch (e) {
+                        reject(errorData);
+                    }
+                });
+            }, this));
+        },
+
+        /**
+         * Loads model data of a root location
+         *
+         * @method _loadRootLocationModelData
+         * @protected
+         * @param rootLocationId {String} root location REST id
+         * @return {Y.Promise}
+         */
+        _loadRootLocationModelData: function (rootLocationId) {
+            var rootLocation = this.get('rootLocation'),
+                errorData = {
+                    message: this.get('loadLocationModelErrorMessage'),
+                    identifier: this.get('loadLocationModelErrorIdentifier')
+                };
+
+            rootLocation.set('id', rootLocationId);
+
+            return new Y.Promise(Y.bind(function (resolve, reject) {
+                rootLocation.load({api: this.get('host').get('capi')}, function (error, response) {
+                    if (error) {
+                        reject(errorData);
+
+                        return;
+                    }
+
+                    resolve(response.document.Location);
+                });
+            }, this));
+        },
+
+        /**
+         * Redirects user to content edit
+         *
+         * @method _goToContentEdit
+         * @protected
+         * @param event {Object} event facade
+         * @param event.contentId {String} REST content id
+         * @param event.languageCode {String} content language
+         */
+        _goToContentEdit: function (event) {
+            this.get('host').get('app').navigateTo('editContent', {
```

Comment:
I guess it's time to have something a bit more generic. We have https://github.com/ezsystems/PlatformUIBundle/blob/master/Resources/public/js/views/services/ez-navigationhubviewservice.js#L25 in navigation hub view service, I think you should move this system to the app so that any component can fire `navigateTo` if there's a need to navigate somewhere. (Also keep the `_navigateTo` method in the navigation hub view service but add in it a deprecation warning and add `@deprecated` tag in the doc).